### PR TITLE
1657395: Add descriptions for SEO (3 of 3)

### PR DESCRIPTION
**Pull request by TimShererWithAquent**
## Summary

We're adding descriptions to the metadata to improve SEO. We're only touching articles that had more than 100 page views in the last month.

Please review. Our SLA specifies 5 working days for review, but let me know if you  need more time. All three PRs total 43 descriptions added. 

Suggestions are welcome, but keep in mind our 100-160 char limit. 

Fixes [1657395](https://dev.azure.com/mseng/TechnicalContent/_workitems/edit/1657395)

@gewarren - You're listed as author for the directories. Please help me find the right person to approve if it isn't you.

### CODE REVIEW

File: docs/framework/windows-workflow-foundation/samples/document-approval-process.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "Document Approval Process"
+description: This sample demonstrates many Windows Workflow Foundation and Windows Communication Foundation features in a document approval process scenario.
```
Comment (gewarren) [inline docs/framework/windows-workflow-foundation/samples/document-approval-process.md:3]:
```suggestion
description: This sample demonstrates many Windows Workflow Foundation and Windows Communication Foundation features in a document-approval process scenario.
```

File: docs/framework/windows-workflow-foundation/state-machine-workflows.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "State Machine Workflows"
+description: Learn about creating state machine workflows using the StateMachine activity, with State, Transition, and others, to build state machine workflow programs.
```
Comment (gewarren) [inline docs/framework/windows-workflow-foundation/state-machine-workflows.md:3]:
```suggestion
description: Learn about creating state machine workflows using the StateMachine activity.
```

File: docs/framework/windows-workflow-foundation/using-workflowinvoker-and-workflowapplication.md [hunk 3]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "Using WorkflowInvoker and WorkflowApplication"
+description: This article describes workflow hosting with WorkflowInvoker and WorkflowApplication, two of the methods of hosting that Windows Workflow Foundation provides.
```
Comment (gewarren) [inline docs/framework/windows-workflow-foundation/using-workflowinvoker-and-workflowapplication.md:3]:
```suggestion
description: This article describes workflow hosting using WorkflowInvoker and WorkflowApplication.
```

File: docs/framework/windows-workflow-foundation/variables-and-arguments.md [hunk 4]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "Variables and Arguments"
+description:This article describes variables, which represent storage of data, and arguments, which represent flow of data to/from an activity in Workflow Foundation.
```
Comment (gewarren) [inline docs/framework/windows-workflow-foundation/variables-and-arguments.md:3]:
```suggestion
description: This article describes variables, which represent storage of data, and arguments, which represent flow of data to/from an activity in Workflow Foundation.
```

File: docs/framework/windows-workflow-foundation/whats-new.md [hunk 5]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "What's New in Windows Workflow Foundation"
+description: Learn about Windows Workflow Foundation in .NET Framework 4 changes to several development paradigms from previous versions.
```
Comment (gewarren) [inline docs/framework/windows-workflow-foundation/whats-new.md:3]:
```suggestion
description: Learn about Windows Workflow Foundation changes in .NET Framework 4.
```

File: docs/framework/windows-workflow-foundation/workflow-persistence.md [hunk 6]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "Workflow Persistence"
+description: The .NET Framework 4.6.1 includes the SqlWorkflowInstanceStore class, which allows persistence of workflow data and metadata into a SQL Server database.
```
Comment (gewarren) [inline docs/framework/windows-workflow-foundation/workflow-persistence.md:3]:
```suggestion
description: .NET Framework 4.6.1 includes the SqlWorkflowInstanceStore class, which allows persistence of workflow data and metadata into a SQL Server database.
```

**Review by gewarren — APPROVED** [review]

File: docs/framework/windows-workflow-foundation/state-machine-workflows.md (hunk 2 above)
Comment (TimShererWithAquent) [reply docs/framework/windows-workflow-foundation/state-machine-workflows.md:3]: @gewarren - this one comes in under 100 char - and we'll bend that rule - you have final say.

Is this OK:

This article provides an overview of creating state machine workflows using the StateMachine activity.

File: docs/framework/windows-workflow-foundation/using-workflowinvoker-and-workflowapplication.md (hunk 3 above)
Comment (TimShererWithAquent) [reply docs/framework/windows-workflow-foundation/using-workflowinvoker-and-workflowapplication.md:3]: @gewarren - How about this:

This article describes workflow hosting using WorkflowInvoker and WorkflowApplication in Windows Workflow Foundation.

File: docs/framework/windows-workflow-foundation/whats-new.md (hunk 5 above)
Comment (TimShererWithAquent) [reply docs/framework/windows-workflow-foundation/whats-new.md:3]: @gewarren - and this: 

Learn about Windows Workflow Foundation changes in .NET Framework 4. Workflows are easier to create, execute, and maintain, and implement new functionality.

File: docs/framework/windows-workflow-foundation/state-machine-workflows.md (hunk 2 above)
Comment (gewarren) [reply docs/framework/windows-workflow-foundation/state-machine-workflows.md:3]: Yes, that sounds great.

File: docs/framework/windows-workflow-foundation/using-workflowinvoker-and-workflowapplication.md (hunk 3 above)
Comment (gewarren) [reply docs/framework/windows-workflow-foundation/using-workflowinvoker-and-workflowapplication.md:3]: Sounds good.

File: docs/framework/windows-workflow-foundation/whats-new.md (hunk 5 above)
Comment (gewarren) [reply docs/framework/windows-workflow-foundation/whats-new.md:3]:
```suggestion
description: Learn about Windows Workflow Foundation changes in .NET Framework 4. Workflows are easier to create, execute, and maintain.
```

**Review by gewarren — APPROVED** [review]

**Pull request merged**